modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/upgrade/OnboardingUpgradeScreen.kt

```diff
@@ -253,19 +253,30 @@ private fun UpgradeContent(
     pages: List<UpgradePagerContent>,
     modifier: Modifier = Modifier,
 ) {
-    Column(modifier = modifier) {
-        val pagerState = rememberPagerState(initialPage = 0) { pages.size }
-        val coroutineScope = rememberCoroutineScope()
-        VerticalPager(state = pagerState, modifier = Modifier.fillMaxSize()) { page ->
-            when (val currentPage = pages[page]) {
+    val coroutineScope = rememberCoroutineScope()
+    val listState = rememberLazyListState()
+    FadedLazyColumn(
+        modifier = modifier,
+        state = listState,
+    ) {
+        itemsIndexed(pages) { index, page ->
+            when (page) {
                 is UpgradePagerContent.Features -> FeaturesContent(
-                    features = currentPage,
-                    onCtaClick = { coroutineScope.launch { pagerState.animateScrollToPage(pages.size - page) } },
+                    features = page,
+                    onCtaClick = {
+                        coroutineScope.launch {
+                            listState.scrollToItem(pages.size - index)
```

Comment:
This has a bit of jarring effect when used. `animateScrollToItem` would help with that.